examples/src/lib.rs

```diff
@@ -123,18 +123,13 @@ mod test {
             let path = path.unwrap();
 
             let contents = read_to_string(path.join("rust-toolchain")).unwrap();
-            let document = contents.parse::<DocumentMut>().unwrap();
-            let array = document
-                .as_table()
+            let table = toml::from_str::<toml::Table>(&contents).unwrap();
+            let components = table
                 .get("toolchain")
-                .and_then(Item::as_table)
-                .and_then(|table| table.get("components"))
-                .and_then(Item::as_array)
-                .unwrap();
-            let components = array
-                .iter()
-                .map(Value::as_str)
-                .collect::<Option<Vec<_>>>()
+                .and_then(|value| value.as_table())
+                .and_then(|toolchain| toolchain.get("components"))
+                .and_then(|value| value.as_array())
+                .map(|array| array.iter().filter_map(|v| v.as_str()).collect::<Vec<_>>())
```

Comment:
```suggestion
            let components = array
                .iter()
                .map(Value::as_str)
                .collect::<Option<Vec<_>>>()
                .unwrap();
```
I think I extracted the components separately on purpose. The reason is: if `as_str` returns `None` for any value in the array, then the overall result will be `None` and the `unwrap` will fail (indicating that there is a problem).

Note that the `filter_map` approach would silently ignore `None`s returned `as_str`.

I think I prefer the former, loud and in-your-face approach.